src/pages/TrendingPage.tsx

```diff
@@ -26,247 +26,297 @@ import { useNavigate } from "react-router-dom";
 import { SearchPluginsPageQuery } from "./SearchPluginsPage";
 import { TorrClient } from "../utils/TorrClient";
 
+
 export interface TrendingPageProps {}
 
 const smallImage = "http://image.tmdb.org/t/p/w200";
 const originalImage = "http://image.tmdb.org/t/p/original";
 
+
 const TrendingPage = (props: TrendingPageProps) => {
-  const tabs = ["Movies", "TV"];
-  const [tab, setTab] = useState(0);
+    const tabs = ["Movies", "TV", "TOP 100"];
+    const [tab, setTab] = useState(0);
 
-  const [selectedMovie, setSelectedMovie] = useState<MovieResult>();
-  const movieBottomSheet = useDisclosure();
-  const { data: trendingMovies } = useQuery("getTrendingMovies", async () =>
-    tmdbClient.trending({
-      media_type: "movie",
-      time_window: "day",
-    })
-  );
+    const [selectedMovie, setSelectedMovie] = useState<MovieResult>();
+    const movieBottomSheet = useDisclosure();
+    const { data: trendingMovies } = useQuery("getTrendingMovies", async () =>
+        tmdbClient.trending({
+            media_type: "movie",
+            time_window: "day",
+        })
+    );
 
-  const {
-    data: TorrData,
-    mutate: getTorrs,
-    isLoading: torrsLoading,
-  } = useMutation((search: string) =>
-    YTSClient.search({
-      query_term: search,
-    })
-  );
+    const [currentPage, setCurrentPage] = useState(1);
 
-  const tvBottomSheet = useDisclosure();
-  const [selectedTv, setSelectedTv] = useState<TvResult>();
-  const { data: trendingTv } = useQuery("getTrendingTv", async () =>
-    tmdbClient.trending({
-      media_type: "tv",
-      time_window: "day",
-    })
-  );
+    const [topMoviesData, setTopMoviesData] = useState<MovieResult[]>([]);
 
-  const { data: plugins } = useQuery(
-    SearchPluginsPageQuery,
-    TorrClient.getInstalledPlugins
-  );
+    const {refetch} = useQuery("getTopMovies", async () => {
+            const res = await tmdbClient.movieTopRated({
+                page: currentPage,
+            })
+            setCurrentPage(currentPage + 1);
+            setTopMoviesData([...topMoviesData, ...res?.results ?? []]);
```

Comment:
Not a big deal, but if you're interested, react-query can handle paginated queries internally
https://tanstack.com/query/v4/docs/react/guides/paginated-queries